pkgs/top-level/all-packages.nix

```diff
@@ -5619,8 +5619,8 @@ with pkgs;
     (lib.addMetaAttrs { outputsToInstall = [ "jre" ]; }
       (openjdk7.jre // { outputs = [ "jre" ]; }));
 
-  jdk8 = openjdk8 // { outputs = [ "out" ]; };
-  jre8 = lib.setName "openjre-${lib.getVersion pkgs.openjdk8.jre}"
+  jdk8 = if stdenv.isArm then oraclejdk8 else openjdk8 // { outputs = [ "out" ]; };
+  jre8 = if stdenv.isArm then oraclejre8 else lib.setName "openjre-${lib.getVersion pkgs.openjdk8.jre}"
```

Comment:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/bootstrap.nix#L32

Actually, there's no bootstrap for openjdk on ARM. I think oraclejdk is a sensible default until someone invests the time to get openjdk building on ARM. It won't install unless they have `allowUnfree = true`, anyway.

The dropbox-hosted bootstraps are a clear indicator of the sort of rabbit-hole I'm indisposed to explore, at the moment. If you really think having a nonfree dist as the default is worse than having no runtime at all for any java packages, then I can remove these lines, but I fail to see what the harm could be.